File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/timers/timer/vb/Timer.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Threading
Imports System.Timers

Namespace Timer

    Module Timer

        Public Sub Main()
            Dim aTimer As System.Timers.Timer
            aTimer = New System.Timers.Timer

            AddHandler aTimer.Elapsed, AddressOf OnTimer

            aTimer.Interval = 1000
            aTimer.Enabled = True

            Console.WriteLine("Press 'q' to quit the sample")
            While (Console.Read() <> 113)
            End While
        End Sub

        Public Sub OnTimer(ByVal source As Object, ByVal e As ElapsedEventArgs)
            Console.WriteLine("Hello World!")
        End Sub

    End Module

End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/timers/timer/cs/timer.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Threading;
using System.Timers;

public class WaitableTimerSample
{
    public static void Main(String[] args)
	{
		System.Timers.Timer aTimer = new System.Timers.Timer();

		aTimer.Elapsed += new ElapsedEventHandler(OnTimer);

		aTimer.Interval = 1000;
		aTimer.Enabled = true;

		Console.WriteLine("Press \'q\' to quit the sample");
		while(Console.Read()!='q');
    }

	public static void OnTimer(Object source, ElapsedEventArgs e)
	{
		Console.WriteLine("Hello World!");
	}
}